packages/transaction/src/index.ts

```diff
@@ -0,0 +1,182 @@
+    const hasCorrectHashLength = isTrytesOfExactLength(hash, HASH_SIZE)
+
+    if (mwm) {
+        return (
+            hasCorrectHashLength &&
+            trytesToTrits(hash)
+                .slice(-Math.abs(mwm))
+                .every(trit => trit === 0)
+        )
+    }
+
+    return hasCorrectHashLength
+}
+
+/**
+ * Checks if input is array of valid transaction hashes.
+ *
+ * @method isTransactionHashArray
+ *
+ * @param {string[]} hashes
+ *
+ * @return {boolean}
+ */
+export const isTransactionHashArray = (hashes: ReadonlyArray<any>): hashes is ReadonlyArray<Hash> =>
+    isArray(hashes) && !!hashes.length && hashes.every(isTransactionHash)
+
+/**
+ * Checks if input is correct transaction trytes (2673 trytes)
+ *
+ * @method isTransactionTrytes
+ *
+ * @param {string} trytes
+ * @param {number} mwm
+ *
+ * @return {boolean}
+ */
+export const isTransactionTrytes = (trytes: any, mwm?: number): trytes is Trytes => {
+    const hasCorrectTrytesLength = isTrytesOfExactLength(trytes, TRANSACTION_TRYTES_SIZE)
+    const isNotNineTrytes = !isNinesTrytes(trytes)
+
+    if (mwm) {
+        return (
+            hasCorrectTrytesLength && isNotNineTrytes && isTransactionHash(transactionHash(trytesToTrits(trytes)), mwm)
+        )
+    }
+
+    return hasCorrectTrytesLength && isNotNineTrytes
+}
+
+/**
+ * Checks if input is array of valid transaction trytes.
+ *
+ * @method isTransactionTrytesArray
+ *
+ * @param {string[]} trytesArray
+ *
+ * @return {boolean}
+ */
+export const isTransactionTrytesArray = (trytesArray: ReadonlyArray<any>): trytesArray is ReadonlyArray<Trytes> =>
+    isArray(trytesArray) &&
+    !!trytesArray.length &&
+    trytesArray.every(trytes => isTrytes(trytes, TRANSACTION_TRYTES_SIZE))
+
+/**
+ * Checks if input is array of valid attached transaction trytes.
+ * For attached transactions last 241 trytes are non-zero.
+ *
+ * @method isAttachedTrytesArray
+ *
+ * @param {string[]} trytesArray
+ *
+ * @return {boolean}
+ */
+export const isAttachedTrytesArray = (trytesArray: ReadonlyArray<any>): trytesArray is ReadonlyArray<Trytes> =>
+    isArray(trytesArray) &&
+    trytesArray.length > 0 &&
+    trytesArray.every(
+        trytes =>
+            isTrytesOfExactLength(trytes, TRANSACTION_TRYTES_SIZE) &&
+            !/^[9]+$/.test(trytes.slice(TRANSACTION_TRYTES_SIZE - 3 * HASH_SIZE))
+    )
+
+/**
+ * Checks if input is valid transaction object.
+ *
+ * @method isTransaction
+ *
+ * @param {object} tx
+ *
+ * @return {boolean}
+ */
+export const isTransaction = (tx: any): tx is Transaction =>
+    isHash(tx.hash) &&
+    isTrytesOfExactLength(tx.signatureMessageFragment, SIGNATURE_MESSAGE_FRAGMENT_TRYTES_SIZE) &&
+    isHash(tx.address) &&
+    isInteger(tx.value) &&
+    isTrytesOfExactLength(tx.obsoleteTag, OBSOLETE_TAG_TRYTES_SIZE) &&
+    isInteger(tx.timestamp) &&
+    (isInteger(tx.currentIndex) && tx.currentIndex <= tx.lastIndex) &&
```

Comment:
That's a nice idea, but I'm not sure if this check should be responsibility of bundle validator or tx validator...
If not we should also add:
```js
tx.currentIndex >= 0
```

Reply:
All right. Added this validation `tx.currentIndex >= 0` in `45c7515`